etf-sdk/src/lib.rs

```diff
@@ -1,237 +1,246 @@
-//     //                     assert_eq!(m, b"test".to_vec());
-//     //                 }, 
-//     //                 Err(_) => {
-//     //                     panic!("test should pass, but decryption failed");
-//     //                 }
-//     //             }
-//     //         },
-//     //         Err(_) => {
-//     //             panic!("test should pass, but encryption failed");
-//     //         }
-//     //     }
-//     // }
-// }
\ No newline at end of file
+use ark_serialize::CanonicalSerialize;
+use serde::Deserialize;
+#[cfg_attr(tarpaulin, skip)]
+use wasm_bindgen::prelude::*;
+use etf_crypto_primitives::{self, encryption::tlock::{DecryptionResult, SecretKey, TLECiphertext}, ibe::fullident::{IBESecret, Identity}, utils::{self, *}};
+use w3f_bls::{EngineBLS, TinyBLS377};
+use rand_chacha::ChaCha20Rng;
+use ark_std::rand::SeedableRng;
+use w3f_bls::{
+    DoublePublicKey,
+    DoublePublicKeyScheme,
+};
+use serde::Serialize;
+use serde_big_array::BigArray;
+use ark_serialize::CanonicalDeserialize;
+
+ #[wasm_bindgen]
+pub fn encrypt( 
+    id_js: JsValue,
+    message_js: JsValue, // &[u8]
+    sk_js: JsValue,
+    p_pub_js: JsValue
+) -> Result<JsValue, JsError> {
+    // msk => master secret key
+    let msk_bytes: [u8;32] = serde_wasm_bindgen::from_value(sk_js.clone())
+        .map_err(|_| JsError::new("could not decode secret key"))?;
+    let mut rng: ChaCha20Rng = ChaCha20Rng::from_seed(msk_bytes);
+    let msk = convert_from_bytes::<<TinyBLS377 as EngineBLS>::Scalar,32>(&msk_bytes.clone()).ok_or(JsError::new("Could not convert secret key"))?;
+    let secret_key = SecretKey::<TinyBLS377>(msk);
+    // Look into String Formatting
+    let pp_conversion: Vec<u8> = serde_wasm_bindgen::from_value(p_pub_js.clone())
+        .map_err(|_| JsError::new("could not decode p_pub"))?;
+    let pp_bytes: [u8;144] = pp_conversion.try_into().map_err(|_| JsError::new("could not convert public params"))?;
+    let double_pub_key = convert_from_bytes::<DoublePublicKey<TinyBLS377>, 144>(&pp_bytes.clone()).ok_or(JsError::new("Could not convert secret key"))?;
+    let pp = double_pub_key.1;
+    let id_bytes: Vec<u8> = serde_wasm_bindgen::from_value(id_js.clone())
+        .map_err(|_| JsError::new("could not decode id"))?;
+    let identity = Identity::new(&id_bytes);
+    let message_bytes: Vec<u8> = serde_wasm_bindgen::from_value(message_js.clone())
+        .map_err(|_| JsError::new("could not decode message"))?;
+    let mut ciphertext_bytes: Vec<_> = Vec::new();
+    let ciphertext = secret_key.encrypt(pp, &message_bytes, identity, &mut rng).map_err(|_| JsError::new("encryption has failed"))?;
+    ciphertext.serialize_compressed(&mut ciphertext_bytes).map_err(|_| JsError::new("ciphertext serialization has failed"))?;
+    
+    return serde_wasm_bindgen::to_value(&ciphertext_bytes).map_err(|_| JsError::new("could not convert ciphertext to JsValue"));
```

Comment:
you don't need the return here, you can remove the 'return' and ';'

Reply:
This has been updated.